hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/SCMDBInsightEndPoint.java

```diff
@@ -0,0 +1,236 @@
+        deletedContainerInfo.setUsedBytes(containerInfo.getUsedBytes());
+        deletedContainerInfo.setReplicationConfig(
+            containerInfo.getReplicationConfig());
+        deletedContainerInfo.setReplicationFactor(
+            containerInfo.getReplicationFactor().name());
+        deletedContainerInfoList.add(deletedContainerInfo);
+      });
+    } catch (IllegalArgumentException e) {
+      throw new WebApplicationException(e, Response.Status.BAD_REQUEST);
+    } catch (Exception ex) {
+      throw new WebApplicationException(ex,
+          Response.Status.INTERNAL_SERVER_ERROR);
+    }
+    return Response.ok(deletedContainerInfoList).build();
+  }
+
+  /**
+   * This API returns list of blocks grouped by container state
+   * (OPEN/CLOSING/CLOSED).
+   * {
+   *   "OPEN": [
+   *     {
+   *       "containerId": 100,
+   *       "localIDList": [
+   *         1,
+   *         2,
+   *         3,
+   *         4
+   *       ],
+   *       "localIDCount": 4,
+   *       "txID": 1
+   *     }
+   *   ]
+   * }
+   * @param limit limits the number of records having list of blocks
+   *              grouped by container state (OPEN/CLOSING/CLOSED)
+   * @param prevKey deletedBlocks table key to skip records before prevKey
+   * @return list of blocks grouped by container state (OPEN/CLOSING/CLOSED)
+   */
+  @GET
+  @Path("/blocks/deletePending")
+  public Response getBlocksPendingDeletion(
+      @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam(RECON_QUERY_LIMIT)
+      int limit,
+      @DefaultValue(PREV_BLOCK_ID_PENDING_FOR_DELETION_DEFAULT_VALUE)
+      @QueryParam(RECON_QUERY_PREVKEY) long prevKey) {
+    if (limit < 0 || prevKey < 0) {
+      // Send back an empty response
+      return Response.status(Response.Status.NOT_ACCEPTABLE).build();
+    }
+    Map<String, List<ContainerBlocksInfoWrapper>>
+        containerStateBlockInfoListMap = new HashMap<>();
+    try (
+        Table<Long,
+            StorageContainerDatanodeProtocolProtos.DeletedBlocksTransaction>
+            deletedBlocksTXTable = DELETED_BLOCKS.getTable(this.scmDBStore);
+        TableIterator<Long, ? extends Table.KeyValue<Long,
+            StorageContainerDatanodeProtocolProtos.DeletedBlocksTransaction>>
+            deletedBlocksTableIterator = deletedBlocksTXTable.iterator()) {
+      boolean skipPrevKey = false;
+      Long seekKey = prevKey;
+      if (prevKey > 0) {
+        skipPrevKey = true;
+        Table.KeyValue<Long,
+            StorageContainerDatanodeProtocolProtos.DeletedBlocksTransaction>
+            seekKeyValue =
+            deletedBlocksTableIterator.seek(seekKey);
+        // check if RocksDB was able to seek correctly to the given key prefix
+        // if not, then return empty result
+        // In case of prevKeyPrefix greater than 0, all the keys are returned
+        if (seekKeyValue == null ||
+            (prevKey > 0 &&
+                !seekKeyValue.getKey().equals(prevKey))) {
```

Comment:
I think we should support the case that prevKey doesn't exist.  Say if there is following records in DELETED_BLOCKS table, 
1.  tx = 1
2. tx = 3
if the prevKey is 2 and limit is 1, we should return  tx=3 record.

Reply:
Ok understood.